frontend/src/components/pages/events/SlotDistribution/index.tsx

```diff
@@ -0,0 +1,95 @@
+import { Button, Grid, Typography } from "@material-ui/core";
+import { useState } from "react";
+import CheckboxSelect from "./CheckboxSelect";
+import { addId } from "./helpers";
+
+type Props = {
+  slotDistribution: { category: number[]; availableSlots: number }[];
+  onUpdateSlotDistribution: (slotDist: { category: number[]; availableSlots: number }[]) => void;
+};
+
+/**
+ * Component for the creating a slot distribution, aka a distribution of the available slots between grade years
+ */
+const SlotDistribution: React.FC<Props> = ({ slotDistribution, onUpdateSlotDistribution }) => {
+  const [slotDistWithId, setSlotDistWithId] = useState<{ category: number[]; availableSlots: number; id: number }[]>(
+    addId(slotDistribution)
+  );
+
+  const usedGrades: number[] = ([] as number[])
+    .concat(...slotDistWithId.map((dist) => dist.category))
+    .sort((a, b) => a - b);
```

Comment:
Couldn't this just be
```suggestion
  const usedGrades: number[] = slotDistWithId.map((dist) => dist.category).sort((a, b) => a - b);
```

Reply:
No, that would create a list of lists. Category is a list of grades (number[]), I do it this way to flatten the list.